reporter/html/src/main/java/org/itsallcode/openfasttrace/report/html/view/html/MarkdownSpanConverter.java

```diff
@@ -1,20 +1,64 @@
 package org.itsallcode.openfasttrace.report.html.view.html;
 
+import java.util.List;
+import java.util.regex.Pattern;
+
 final class MarkdownSpanConverter
 {
+    private static final RegexReplacement INDENTED_CODE = RegexReplacement.create("(    .*[\n])+", "<pre>$1</pre>");
+    private static final RegexReplacement BACKTICK_QUOTED_CODE = RegexReplacement.create("`(.*?)`", "<code>$1</code>");
+    private static final RegexReplacement LINK = RegexReplacement.create("\\[([^]]*?)\\]\\(([^)].*?)\\)",
+            "<a href=\"$2\">$1</a>");
+    private static final RegexReplacement BOLD_TEXT = RegexReplacement.create("(__|\\*\\*)(\\p{L}(?:.*\\p{L}))\\1",
+            "<strong>$2</strong>");
+    private static final RegexReplacement EMPHASIZED_TEXT = RegexReplacement.create("([_*])(\\p{L}(?:.*\\p{L}))\\1",
+            "<em>$2</em>");
+
+    private static final List<RegexReplacement> ALL_MARKDOWN_REPLACEMENTS = List.of(INDENTED_CODE, BACKTICK_QUOTED_CODE,
+            LINK, BOLD_TEXT, EMPHASIZED_TEXT);
+
     // Prevent instantiation
     private MarkdownSpanConverter()
     {
     }
 
+    // [impl->dsn~reporting.html.escape-html~1]
     static String convertLineContent(final String input)
     {
-        String text = input;
-        text = text.replaceAll("(    .*[\n])+", "<pre>$1</pre>");
-        text = text.replaceAll("`(.*?)`", "<code>$1</code>");
-        text = text.replaceAll("\\[([^]]*?)\\]\\(([^)].*?)\\)", "<a href=\"$2\">$1</a>");
-        text = text.replaceAll("(__|\\*\\*)(\\p{L}(?:.*\\p{L}))\\1", "<strong>$2</strong>");
-        text = text.replaceAll("([_*])(\\p{L}(?:.*\\p{L}))\\1", "<em>$2</em>");
+        String text = escapeHtml(input);
+        for (final RegexReplacement replacement : ALL_MARKDOWN_REPLACEMENTS)
```

Comment:
:+1: Yep, that is definitively cleaner. I like it!

Sidenote: This is a place for potential later performance improvement. And yes, I am aware, there are libraries for that. :-)

Reply:
This already improves performance by pre-compiling the regular expressions ;)